src/metadata/api.ts

```diff
@@ -33,44 +21,16 @@ export class MetadataApi {
     const { appDataParams, metadataParams } = params || {}
     const { referrerParams, quoteParams, orderClassParams } = metadataParams || {}
 
-    const metadata: latest.Metadata = {}
-    if (referrerParams) {
-      metadata.referrer = createReferrerMetadata(referrerParams)
+    return {
```

Comment:
@alfetopito wdyt about the solution?

Reply:
Nice but it doesn't use `createAppDataDoc` which takes care of the versioning

Also, didn't know you can spread `null` into an object. Neat!

Reply:
Discussed, moved back the previous approach with lazy loading